src/libaktualizr/package_manager/dockerapp_bundles.cc

```diff
@@ -0,0 +1,169 @@
+#include "dockerappmanager.h"
+#include "packagemanagerfactory.h"
+
+#include <sstream>
+
+/**
+ * @brief This package manager compliments the OSTreePackageManager by also including optional Docker Apps.
+ *
+ * A full description of the Docker App project can be found here:
+ *  https://github.com/docker/app/
+ *
+ * Docker Apps are very analogous to docker-compose. In fact, this module
+ * currently "renders" the docker-app file into a docker-compose file. Each
+ * Docker App appears as a custom metadata for a Target in the TUF targets list.
+ *
+ *   "targets": {
+ *    "raspberrypi3-64-lmp-144" : {
+ *      "custom" : {
+ *         "docker_apps" : {
+ *            "httpd" : {
+ *               "uri" :
+ * "hub.foundries.io/andy-corp/shellhttpd@sha256:611052af5d819ea979bd555b508808c8a84f9159130d455ed52a084ab5b6b398"
+ *            }
+ *         },
+ *         "hardwareIds" : ["raspberrypi3-64"],
+ *         "name" : "raspberrypi3-64-lmp",
+ *         "targetFormat" : "OSTREE",
+ *         "version" : "144"
+ *      },
+ *      "hashes" : {"sha256" : "20ac4f7cd50cda6bfed0caa1f8231cc9a7e40bec60026c66df5f7e143af96942"},
+ *      "length" : 0
+ *     }
+ *   }
+ */
+
+struct AppBundle {
+  AppBundle(std::string app_name, const DockerAppManagerConfig &config)
+      : name(std::move(app_name)),
+        app_root(config.docker_apps_root / name),
+        app_params(config.docker_app_params),
+        compose_bin(boost::filesystem::canonical(config.docker_compose_bin).string()) {}
+
+  // Utils::shell isn't interactive. The docker app commands can take a few
+  // seconds to run, so we use std::system to stream it to stdout/sterr
+  bool cmd_streaming(const std::string &cmd) { return std::system(cmd.c_str()) == 0; }
```

Comment:
Is the problem just that the output of `Utils::shell` is only dumped once the call returns? Or why do you need an interactive shell?

No big deal at any rate, this is fine if it does what you need.

Reply:
Its to stream output as the command runs. In the original implementation, I'd copy and pasted that logic around and had to keep that comment. So for this new one I decided to put it in one place so it wasn't as confusing.